src/main/java/datawave/microservice/query/storage/QueryStorageCacheImpl.java

```diff
@@ -0,0 +1,493 @@
+            }
+        } finally {
+            lock.unlock();
+        }
+        return false;
+    }
+    
+    /**
+     * Get a lock for the query status
+     *
+     * @param queryId
+     *            the query id
+     * @return a lock object
+     */
+    @Override
+    public QueryStorageLock getQueryStatusLock(String queryId) {
+        return queryStatusCache.getQueryStatusLock(queryId);
+    }
+    
+    /**
+     * Get a task states lock
+     *
+     * @param queryId
+     *            the query id
+     * @return a lock object
+     */
+    @Override
+    public QueryStorageLock getTaskStatesLock(String queryId) {
+        return taskStatesCache.getTaskStatesLock(queryId);
+    }
+    
+    /**
+     * Get the current task states.
+     *
+     * @param queryId
+     *            the query id
+     * @return the task states
+     */
+    @Override
+    public TaskStates getTaskStates(String queryId) {
+        return taskStatesCache.getTaskStates(queryId);
+    }
+    
+    /**
+     * update the query status
+     *
+     * @param taskStates
+     *            the task states
+     */
+    @Override
+    public void updateTaskStates(TaskStates taskStates) {
+        taskStatesCache.updateTaskStates(taskStates);
+    }
+    
+    /**
+     * Create a new query task. This will create a new query task, store it.
+     * 
+     * @param action
+     *            The query action
+     * @param checkpoint
+     *            The query checkpoint
+     * @return The new query task
+     */
+    @Override
+    public QueryTask createTask(QueryRequest.Method action, QueryCheckpoint checkpoint) {
+        QueryTask task;
+        
+        String queryId = checkpoint.getQueryKey().getQueryId();
+        
+        QueryStorageLock lock = taskStatesCache.getTaskStatesLock(queryId);
+        lock.lock();
+        try {
+            // Set the initial ready state in the task states
+            TaskStates states = taskStatesCache.getTaskStates(queryId);
+            
+            int taskId = states.getAndIncrementNextTaskId();
+            
+            // create a query task in the cache
+            task = taskCache.addQueryTask(taskId, action, checkpoint);
+            
+            states.setState(taskId, TaskStates.TASK_STATE.READY);
+            
+            taskStatesCache.updateTaskStates(states);
+        } catch (Exception e) {
+            log.error("Failed to add query task", e);
+            throw new RuntimeException("Failed to add query task", e);
+        } finally {
+            lock.unlock();
+        }
+        
+        // return the task
+        return task;
+    }
+    
+    /**
+     * Get a task for a given task key and lock it for processing. This return null if the task no longer exists. This will throw an exception if the task is
+     * already locked.
+     *
+     * @param taskKey
+     *            The task key
+     * @return The query task, null if deleted
+     * @throws TaskLockException
```

Comment:
no exceptions thrown here, remove the annotation

Reply:
good catch.  i didn't see this exception being thrown anywhere actually, so i deleted it.